pkg/databaseclaim/requestinfo.go

```diff
@@ -0,0 +1,95 @@
+package databaseclaim
+
+import (
+	"fmt"
+	v1 "github.com/infobloxopen/db-controller/api/v1"
+	basefun "github.com/infobloxopen/db-controller/pkg/basefunctions"
+	"github.com/infobloxopen/db-controller/pkg/hostparams"
+)
+
+// requestInfo is a struct that holds the information needed to create a database.
+type requestInfo struct {
+	DbType                     v1.DatabaseType
+	SharedDBHost               bool
+	MasterConnInfo             v1.DatabaseClaimConnectionInfo
+	TempSecret                 string
+	DbHostIdentifier           string
+	HostParams                 hostparams.HostParams
+	EnableReplicationRole      bool
+	EnableSuperUser            bool
+	EnablePerfInsight          bool
+	EnableCloudwatchLogsExport []*string
+	BackupRetentionDays        int64
+	CACertificateIdentifier    string
+}
+
+// setRequestInfo load base values and configs to kick off the database claim request.
+func (r *DatabaseClaimReconciler) setRequestInfo(dbClaim *v1.DatabaseClaim) (requestInfo, error) {
```

Comment:
The goal is to split all of this code up into smaller pieces. To get there, this should not hang off the reconciler struct and make a bunch of calls into the reconciler to find things. 

Change this to receive the Config and dbClaim as parameters. Made some comments above on getDynamicHostName to remove that dependence on reconciler struct too.
```suggestion
func NewRequestInfo(config *DatabaseClaimConfig, dbClaim *v1.DatabaseClaim) (requestInfo, error) {
```